### PR TITLE
BUG: void dtype setup checked offset not actual pointer for alignment

### LINKED ISSUE #11088: 1.14.3 on sparc64: numpy.core.tests.test_multiarray.TestMethods.test__complex__should_not_work ... Bus error
Hello,
when building numpy 1.14.3 on debian for sparc64 architecture, the test suite fails with:

```
numpy.core.tests.test_multiarray.TestMatmul.test_shapes ... ok
numpy.core.tests.test_multiarray.TestMatmul.test_vector_matrix_values ... ok
numpy.core.tests.test_multiarray.TestMatmul.test_vector_vector_values ... ok
numpy.core.tests.test_multiarray.TestMemEventHook.test_mem_seteventhook ... ok
numpy.core.tests.test_multiarray.TestMethods.test__complex__ ... ok
numpy.core.tests.test_multiarray.TestMethods.test__complex__should_not_work ... Bus error
```

(this is for python 2.7.15~rc1)

full log at: https://buildd.debian.org/status/fetch.php?pkg=python-numpy&arch=sparc64&ver=1%3A1.14.3-2&stamp=1525577504&raw=0

can you have a look? thanks!

### PR DESCRIPTION
**Pull request by seberg**
This commit also adds an assert and closes gh-11088

---
Ok, the assert, its still in there with sizeof, which is probably OK here, but not right generally....

### CODE REVIEW

**Comment by seberg** [comment]
OK, silly me, the problem was that the macro is defined in the local file, and it was defined below... and I missed the first error ;).

**Comment by seberg** [comment]
Maybe the macro should be moved all the way up, it is probably pretty random there.

File: numpy/core/src/multiarray/arraytypes.c.src [hunk 1]
```diff
@@ -760,7 +771,8 @@ _setup_field(int i, PyArray_Descr *descr, PyArrayObject *arr,
     }
 
     ((PyArrayObject_fields *)(arr))->descr = new;
-    if ((new->alignment > 1) && ((offset % new->alignment) != 0)) {
+    if ((new->alignment > 1) && (
```
Comment (charris) [inline numpy/core/src/multiarray/arraytypes.c.src]: The `(` would look better on the next line.

Comment (charris) [reply numpy/core/src/multiarray/arraytypes.c.src]: Committer can fix later if needed.

**Comment by charris** [comment]
LGTM.

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
Thanks Sebastian.

**Pull request merged**